device-registry/src/main/scala/org/genivi/sota/device_registry/SystemInfoResource.scala

```diff
@@ -45,8 +46,12 @@ class SystemInfoResource(deviceNamespaceAuthorizer: Directive1[Uuid])
     f
   }
 
-  def fetchSystemInfo(uuid: Uuid): Route =
-    complete(db.run(SystemInfoRepository.findByUuid(uuid)))
+  def fetchSystemInfo(uuid: Uuid): Route = {
+    val comp = db.run(SystemInfoRepository.findByUuid(uuid)).recover{
+      case MissingEntity(x) if x == classOf[SystemInfoRepository.SystemInfo] => Json.obj()
```

Comment:
we should never compare classes like this. we should match on the real class, MissingSystemInfo in this case.